genienlp/tasks/generic_dataset.py

```diff
@@ -1929,3 +1929,50 @@ def return_splits(cls, name, path='.data', train='train', validation='dev', test
         return Split(train=all_train_data, eval=all_validation_data, test=all_test_data), Split(
             train=train_path, eval=validation_path, test=test_path
         )
+
+
+class OODDataset(CQA):
+    name = 'ood'
+
+    def __init__(self, path, lower=False, cached_path=None, skip_cache=False, **kwargs):
+        examples = []
+        labels = {'neg': 'negative', 'pos': 'positive'}
+        question = 'Is this sentence in-domain or out-domain?'
+
+        cache_name = os.path.join(cached_path, os.path.basename(path))
+        if os.path.exists(cache_name) and not skip_cache:
+            logger.info(f'Loading cached data from {cache_name}')
+            examples = torch.load(cache_name)
+        else:
+            for label in ['pos', 'neg']:
```

Comment:
Not sure why you need two separate files. The last column of your files contains this information (it's pos if it has thingtalk, otherwise it's ood)